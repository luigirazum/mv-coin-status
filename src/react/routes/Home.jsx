import { Link } from 'react-router-dom';

const Home = () => (
  <main>
    <section className="home">
      <div className="wrapper">
        <h1>Welcome to CoinStatus</h1>
        <div>
          <p>Here you will find the most current information about your favorite crypto coin.</p>
          <p>Press the Start button below to explore the world of cryptocurrencies.</p>
        </div>
        <Link
          to="coins"
          className="fsControls btn btn-home btn-start"
        >
          Start
        </Link>
      </div>
    </section>
  </main>
);

export default Home;
