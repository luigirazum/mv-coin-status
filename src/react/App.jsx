import { Routes, Route } from 'react-router-dom';
import Home from './routes/Home';
import CoinsLayout from './routes/CoinsLayout';
import AllCoins from './routes/AllCoins';
import CoinDetails from './routes/CoinDetails';
import '../styles/App.css';

const App = () => (
  <Routes>
    <Route index element={<Home />} />
    <Route path="coins" caseSensitive element={<CoinsLayout />}>
      <Route index caseSensitive element={<AllCoins />} />
      <Route path=":id" caseSensitive element={<CoinDetails />} />
    </Route>
    <Route path="*" element={<h3>ERROR</h3>} />
  </Routes>
);

export default App;
