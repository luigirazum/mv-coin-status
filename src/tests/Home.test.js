import { screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import renderWithProvider from '../react/utils/renderUtils';
import Home from '../react/routes/Home';

let container = null;
let testRender = null;
let queryByText;
let queryByRole;

const resetTestEnv = () => {
  container.remove();
  container = null;
  testRender = null;
};

beforeEach(() => {
  // setup a DOM element as a render target
  testRender = renderWithProvider(
    <Home />,
    {},
    { routerWrapper: { memory: true } },
  );
  ({ container, queryByText } = testRender);
  document.body.appendChild(container);
});

afterEach(() => {
  //  cleanup on exiting
  resetTestEnv();
});

/* START: Home page component test */
describe('Home component tests', () => {
  /* START: Home page render */
  describe('Home page must be rendered', () => {
    test('should render a Welcome', () => {
      expect(queryByText('Welcome')).toBeInTheDocument();
    });
    test('should render a Start button', () => {
      expect(queryByText('Start')).toBeInTheDocument();
    });
    test('should render an About button', () => {
      expect(queryByText('About')).toBeInTheDocument();
    });
  });
  /* END: Home page render */

  describe('Home page component snapshot test', () => {
    test('snapshot renders correctly', () => {
      const { component } = testRender;

      const tree = create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
/* END: Coin component tests */
