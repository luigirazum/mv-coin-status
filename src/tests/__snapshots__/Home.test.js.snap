// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component tests Home page component snapshot test snapshot renders correctly 1`] = `
<main>
  <h1>
    Welcome
  </h1>
  <a
    className="fsControls btn btn-home btn-start"
    href="/coins"
    onClick={[Function]}
  >
    Start
  </a>
  <a
    className="fsControls btn btn-home btn-about"
    href="/"
    onClick={[Function]}
  >
    About
  </a>
</main>
`;
